### DISCUSSION TITLE
Hint before resolving references in a tree

### DISCUSSION THREAD

**Discussion by captain-yoshi** [biojppm/rapidyaml#286] [Ideas]
It would be awesome if we could query the `ryml::Tree` to see whether it encountered an anchor or an alias when parsing. By doing so, the user could have a hint to resolve or not the tree and thus skip the full traversal of the tree if there is no references.

``` cpp
Tree tree;
tree = ryml::parse_in_arena(buf)

// resolve if potential references
if(tree.has_ref())
    tree.resolve()
```
Maybe the API already does this but I did not find any way without traversing the tree.

**Comment by biojppm** [comment]
This is a good idea. I'll eventually get to this.
